src/commands/misc_commands/news.ts

```diff
@@ -0,0 +1,443 @@
+        const jobName = scheduledJobName(
+            subscription.guildID,
+            subscription.textChannelID,
+            subscription.range,
+        );
+
+        if (subscription.range === NewsRange.DAY) {
+            schedule.scheduleJob(jobName, "0 0 * * *", async () => {
+                await NewsCommand.sendNews(
+                    subscriptionContext,
+                    subscription.range,
+                );
+            });
+        } else if (subscription.range === NewsRange.WEEK) {
+            schedule.scheduleJob(jobName, "0 0 * * 0", async () => {
+                await NewsCommand.sendNews(
+                    subscriptionContext,
+                    subscription.range,
+                );
+            });
+        }
+    };
+
+    static subscribeNews = async (
+        messageContext: MessageContext,
+        range: NewsRange,
+        interaction: Eris.CommandInteraction,
+    ): Promise<void> => {
+        const subscription: NewsSubscription = {
+            guildID: messageContext.guildID,
+            textChannelID: messageContext.textChannelID,
+            range,
+            createdAt: new Date(),
+        };
+
+        await dbContext.kmq
+            .insertInto("news_subscriptions")
+            .values({
+                guild_id: subscription.guildID,
+                text_channel_id: subscription.textChannelID,
+                range: subscription.range,
+                created_at: subscription.createdAt,
+            })
+            .onDuplicateKeyUpdate({
+                guild_id: subscription.guildID,
+                text_channel_id: subscription.textChannelID,
+                range: subscription.range,
+                created_at: subscription.createdAt,
+            })
+            .execute();
+
+        this.scheduleNewsJob(subscription);
+
+        await sendInfoMessage(messageContext, {
+            title: i18n.translate(
+                messageContext.guildID,
+                "command.news.subscribe.title",
+            ),
+            description: i18n.translate(
+                messageContext.guildID,
+                "command.news.subscribe.description",
+            ),
+        });
+
+        await this.sendNews(messageContext, range, interaction);
+    };
+
+    static unsubscribeNews = async (
+        messageContext: MessageContext,
+        range: NewsRange,
+        interaction: Eris.CommandInteraction,
+    ): Promise<void> => {
+        await sendInfoMessage(
+            messageContext,
+            {
+                title: i18n.translate(
+                    messageContext.guildID,
+                    "command.news.unsubscribe.title",
+                ),
+                description: i18n.translate(
+                    messageContext.guildID,
+                    "command.news.unsubscribe.description",
+                ),
+            },
+            false,
+            undefined,
+            [],
+            interaction,
+        );
+
+        schedule.cancelJob(
```

Comment:
We should check if the guild has subscribed to begin with, otherwise they might think this commmand is doing nothing

Reply:
same with in `subscribeNews`